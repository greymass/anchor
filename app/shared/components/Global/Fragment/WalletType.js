// @flow
import React, { PureComponent } from 'react';
import { Label, Popup } from 'semantic-ui-react';
import { withTranslation } from 'react-i18next';


export class GlobalFragmentWalletType extends PureComponent<Props> {
  render() {
    const {
      mode,
      t
    } = this.props;
    let color = 'grey';
    let icon = 'disk';
    switch (mode) {
      case 'ledger': {
        color = 'purple';
        icon = 'usb';
        break;
      }
      case 'watch': {
        color = 'grey';
        icon = 'eye';
        break;
      }
      case 'cold': {
        color = 'blue';
        icon = 'snowflake';
        break;
      }
      default: {
        color = 'green';
        icon = 'id card';
      }
    }
    return (
      <Popup
        content={t(`wallet:wallet_mode_explain_${mode}_2`)}
        header={t('tools:tools_wallets_mode')}
        inverted
        position="right center"
        trigger={(
          <Label
            basic
            color={color}
            content={t(`global_modal_account_import_${mode}_wallet`)}
            icon={icon}
            position="left center"
          />
        )}
      />
    );
  }
}

export default withTranslation('global')(GlobalFragmentWalletType);
