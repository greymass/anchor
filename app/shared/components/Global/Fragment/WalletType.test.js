import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalFragmentWalletType } from './WalletType';

const mockProps = {
  children: <React.Fragment />,
  content: <React.Fragment />,
  mode: 'hot',
  t: text => text,
};

describe('GlobalFragmentWalletType', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalFragmentWalletType {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
