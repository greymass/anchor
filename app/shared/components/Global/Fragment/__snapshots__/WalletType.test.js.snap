// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFragmentWalletType renders correctly 1`] = `
<Popup
  content="wallet:wallet_mode_explain_hot_2"
  header="tools:tools_wallets_mode"
  inverted={true}
  keepInViewPort={true}
  on="hover"
  position="top left"
  trigger={
    <Label
      basic={true}
      color="green"
      content="global_modal_account_import_hot_wallet"
      icon="id card"
      position="left center"
    />
  }
/>
`;
